lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -479,44 +496,57 @@ def main():
                     rule['from_port'] = None
                     rule['to_port'] = None
 
-                # Convert ip to list we can iterate over
-                if not isinstance(ip, list):
-                    ip = [ip]
-
-                # If rule already exists, don't later delete it
-                for thisip in ip:
-                    ruleId = make_rule_key('in', rule, group_id, thisip)
+                if group_id:
+                    ruleId = make_rule_key('in', rule, group.id, group_id)
                     if ruleId in groupRules:
                         del groupRules[ruleId]
-                    # Otherwise, add new rule
                     else:
-                        grantGroup = None
-                        if group_id:
-                            grantGroup = groups[group_id]
-
                         if not module.check_mode:
-                            group.authorize(rule['proto'], rule['from_port'], rule['to_port'], thisip, grantGroup)
-                        changed = True
-
+                            ipPermissions = createGrantGroupIpPermission(group_id, rule)
+                            if ipPermissions:
+                                ips=ipPermissions
+                                if vpc_id:
+                                    ips=map(lambda x: map(lambda y: y.update({'VpcId': vpc_id}), x.get('UserIdGroupPairs')), [ipPermissions])
+                                try:
+                                    group.authorize_ingress(GroupName=group.group_name, IpPermissions=[ips])
+                                except exceptions.ClientError as e:
+                                    module.fail_json(msg="Unable to authorize ingress for group %s security group '%s' - %s" % (group_id, group, e.message), exception=traceback.format_exc())
```

Comment:
Use the Ansible standard [exception handling guidelines for boto3](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-2)